### ISSUE TITLE
"Duplicate keys in list of rules." in the presence of conditions

### ISSUE THREAD

**Issue by vsht** [WolframResearch/codeinspector#13]
Hi! The following code snippet
```
CodeInspectSummarize[
  "{foo[a_,b_]:>bar1[a,b]/;a=!=b, foo[a_,b_]:>bar2[a,b]/;a===b}"]
```
 triggers a warning, yet the code works as expected. 

Of course one could debate whether the conditionals should be always put before `:>`
or not, but I'm curious if the given example really should be considered problematic
from the technical point of view.

**Comment by bostick** [comment]
This is interesting because of how `Condition` (`/;`) works and where it can be placed.

Technically, even this is fine to do:
```
foo[a_, b_] :> Module[{c}, c = (a === b); bar2[a, b] /; c]
```
The docs for `Condition` mention that.

(Obviously, the use of `Condition` in `Module` probably happens more with function definitions than simple pattern replacements.)

I wonder how much work CodeInspector should do to find the extremely context-sensitive `Condition` that may be buried somewhere. I will have to think about what to do.

**Comment by szhorvat** [comment]
I would think that using `Module` with `:>` (instead of `:=`) is an extremely rare use case. A linter can't avoid false positives and be still useful.

BTW I regularly disable this lint because it always triggers on things like `Graph[{1->2, 1->3}]`.

**Comment by bostick** [comment]
I'm sure a single `/;` at the top, like in the original input, is the vast majority of uses of `/;`. I will take care of this single case for now, and then see if anyone ever complains about the more complicated cases and think of what to do when the time comes.

My philosophy is that it i good to still have some warning for `Graph[{1->2, 1->3}]`, even at very low confidence (0.1), because of copy/paste typos.
